examples/bri-3/src/bri/zkp/agents/proofStorage.agent.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { BlockchainService } from '../../../../blockchain/blockchain.service';
+
+import { Proof } from '../models/proof';
+
+@Injectable()
+export class ProofStorageAgent extends BlockchainService {
+  async storeProofInShieldContract(proof: Proof): Promise<void> {
+    await this.store(proof.payload);
+  }
+
+  async verifyProofInShieldContract(proofToVerify: string): Promise<boolean> {
```

Comment:
would verify method in shield contract be separated from storing the proof? in this version there is verifyAndPush method only https://github.com/eea-oasis/baseline/blob/main/core/contracts/contracts/privacy/Shield.sol
will we use this shield contract or create different one?

Reply:
see my comments above about changing this since we have 2 ways to verify a proof, onchain and offchain